infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutAssetService.kt

```diff
@@ -121,7 +120,7 @@ abstract class LayoutAssetService<ObjectType : LayoutAsset<ObjectType>, DaoType
     private fun verifyInsertResponse(officialId: IntId<ObjectType>?, response: DaoResponse<ObjectType>) {
         if (officialId != null) require(response.id == officialId) {
             "Insert response ID doesn't match object: officialId=$officialId updated=$response"
-        } else require(response.id == response.rowVersion.id) {
+        } else require(response.id.intValue == response.rowVersion.rowId.intValue) {
```

Comment:
Tässä verrataan edelleen id-nimistä kenttää rowId-kenttään, meneekö kuitenkin oikein?

Reply:
Joo, tässä tämä kikkaillaan tarkoituksella. Tässä verifioidaan nimenomaa inserttiä tilanteessa jossa ollaan luomassa uutta draftia (officialId == null). Tällöin käsitteen viralliseksi ID:ksi pitäisi päätyä luodun uuden rivin ID, kun taas yllä (officialId != null) uuden rivin virallinen ID pitäisi olla tuo annettu (käsitteen official ID säilyy). Sinällään tämä on ehkä hiukan turhakin sanity-check, mutta en nyt tässä kohtaa poistanutkaan.